[Feature/#55] 필터 페이지 구현

holdn2 opened the pull request:
## 📌 Related Issues

- close #55 

## ✅ 체크 리스트

- [ ] PR 제목의 형식을 잘 작성했나요? e.g. [Feat/#이슈번호] PR 템플릿 작성
- [ ] 빌드가 성공했나요? (pnpm build)
- [ ] 컨벤션을 지켰나요?
- [ ] 이슈는 등록했나요?
- [ ] 리뷰어와 라벨을 지정했나요?

## 📄 Tasks

- 캘린더 컴포넌트 리팩토링 (커스텀 훅 구현 및 부모 컴포넌트의 state를 prop으로 받도록 수정)
- ButtonText 컴포넌트 밑줄 border 말고 underline으로 수정
- 캘린더 변경 디자인 적용 (지난 날짜들 가운데 라인으로 disable, 범위 내의 색칠 높이 수정)
- 필터 칩 그룹 컴포넌트 구현(중복인지 아닌지에 따라 조건부 렌더링 되도록 처리)
- 필터 페이지 구현 완료 (필터링에 필요한 상태들 하나의 객체로 관리)

## ⭐ PR Point (To Reviewer)

- 우선 필터링할 칩에 대한 내용을 constant 내에 정의해뒀는데 피드백 부탁드립니다.

## 📷 Screenshot

<img width="465" height="1013" alt="image" src="https://github.com/user-attachments/assets/e68fdcc8-ff67-47a1-ae0b-6510140e2614" />
<img width="462" height="1005" alt="image" src="https://github.com/user-attachments/assets/66af92db-85a1-48d2-84c0-7f84217528f4" />

## 🔔 ETC

- 객체 비교를 위해 `lodash` 설치
- 추후 필요하다면 메모이제이션 등으로 최적화할 예정입니다.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - 필터 페이지 추가: 행사 종류, 일정(바텀시트 달력으로 다중 기간 추가/편집), 수량, 음식 종류(다중 선택), 전기 사용, 결제 방법, 초기화/적용 제공
  - 필터 UI 컴포넌트 및 옵션 상수 추가, /filter 경로 등록
- Bug Fixes
  - 바텀시트 열림 시 화면 스크롤 잠금 및 표시 우선순위(z-50) 개선
- Refactor
  - 달력 로직 훅(useCalendar)·유틸(getCalendarDays)로 재구성; 기존 달력 훅 제거 및 컴포넌트 간 책임 정리
- Style
  - 텍스트 버튼에 밑줄 스타일 적용, 달력 일자 크기·비활성 표시 개선, 내비게이션 클래스 정리
- Chores
  - lodash 런타임 및 타입 의존성 추가
- Tests / Docs
  - 버튼 컴포넌트에 'edit' 스타일 스토리 추가
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

holdn2 committed:
feature: 필터 페이지 버플리싱 완료

holdn2 committed:
refactor: Calendar 관련 리팩토링

- state 및 함수 커스텀 훅으로 관리
- 기존 훅은 유틸 함수로 변경

holdn2 committed:
fix: css 관련 수정

holdn2 committed:
style: 캘린더 컴포넌트 요일 margin 수정

holdn2 committed:
Merge branch 'develop' of https://github.com/CHA-CALL/CHA-CALL_FRONT into feature/#55/filter-page

holdn2 committed:
fix: ButtonText 컴포넌트 수정

holdn2 committed:
fix: 캘린더 변경 스타일 적용

holdn2 committed:
feat: 필터 페이지 기능 구현 완료.

필터링 옵션들 객체로 관리

holdn2 committed:
fix: 캘린더 컴포넌트 height 수정

holdn2 committed:
fix: 빌드 오류 수정

holdn2 committed:
code review: 코드래빗 리뷰 반영

holdn2 committed:
code review: 코드래빗 리뷰 반영. 바텀시트 overlay 관련 스타일 수정

hamxxn commented on src/pages/filter/Filter.tsx:
```diff
@@ -0,0 +1,214 @@
+        leftIcon={<Icon name='ic_back' color='#19212A' />}
+        handleLeftClick={() => navigate(-1)}
+        rightIcon={
+          <div className='caption-m-12 text-grayscale-700 border border-grayscale-200 rounded-[0.4rem] px-[1.25rem] py-[0.5rem]'>
```
네비게이션의 righticon을 div으로 바꾸는게 더 좋아보여요!

holdn2 replied:
그럼 제가 네비게이션을 수정해도 될까요?

hamxxn commented on src/pages/filter/Filter.tsx:
```diff
@@ -0,0 +1,214 @@
+    <div>
+      <Navigation
+        leftIcon={<Icon name='ic_back' color='#19212A' />}
+        handleLeftClick={() => navigate(-1)}
```
핸들러는 외부에 정의해주세요!

holdn2 replied:
넵. 반영할게요!

hamxxn commented on src/pages/filter/constant/filter-option-constants.ts:
```diff
@@ -0,0 +1,37 @@
+const EVENT_TYPE = [
```
상수로 뺀거 좋네요!
gitkeep을 추가로 생성하신 이유 있으실까요??

holdn2 replied:
처음 폴더 생성할 때 함께 생성했는데 삭제를 안했네요 ㅎㅎ 삭제하겠습니다.

hamxxn commented on src/pages/filter/Filter.tsx:
```diff
@@ -0,0 +1,214 @@
+
+      <div
+        className={cn(
+          'fixed bottom-[0rem] w-full max-w-[60rem] bg-white px-[2rem] py-[1.7rem]',
```
sticky와 fixed의 차이는 뭘까요??
현재 페이지에선 sticky가 더 맞다고 판단됩니다!

holdn2 replied:
하단 버튼의 경우 항상 화면 아래에 고정인 요소라서 스크롤에 따라 다른 sticky보다 fixed가 맞다고 판단했습니다! sticky는 스크롤을 하며 고정되어야 할 곳에서만 유동적으로 고정되는 속성으로 알고 있어서 fixed로 했습니다! 추가로 피그마에도 fixed라고 해주셔서요. 혹시 제가 잘못이해한 부분이 있다면 알려주세요!

hamxxn replied:
fixed 같은 경우 계속 layout을 만져야해서 저는 선호하진 않아요! fixed를 쓰면 padding, margin을 주고 max-width가 추후에 변경될 경우 하나하나 들어가서 width도 건드려야해서 저는 sticky를 자주 사용하긴 합니다!
찬영님이 더 맞다고 생각하시는 방향으로 선택도 될 것 같아요!

holdn2 replied:
좋은 의견 감사드립니다! 추후 개발에도 참고할게요!

hamxxn reviewed (commented):
pick 사용 너무 좋네요!
저랑 다르게 구현한 부분이 많아 재밌게 봤어요!
현재 모두 선택하지 않았을 경우 bottom button이 클릭이 가능해요!
해당 부분 막아주세요!
-> 한개라도 입력되면 활성화된다고 합니다!

holdn2 committed:
feature: 필터 페이지 일정 추가하기를 누르고 일정을 선택안했을 때는 필터 요청에 포함 안되도록 기능 추가

holdn2 committed:
code review: 코드 리뷰 반영

- 핸들러 외부로 빼기
- Navigation rightIcon div태그로 변경
- Button컴포넌트 edit 추가
- 필요없는 gitkeep 파일 삭제

junghyungo commented on src/pages/filter/Filter.tsx:
```diff
@@ -0,0 +1,227 @@
+      >
+        <Button
+          variant='cta'
+          buttonStyle={isEqual(filters, initialFilter) ? 'disabled' : 'active'}
```
행사 종류, 일정, 음식 종류 등 필터를 하나만 선택하여도 버튼이 `active`가 되는데, 모든 필터를 선택하기 전 까지는 `disabled`로 유지되는건 어떨까요?

holdn2 replied:
해당 부분은 pm님한테 여쭤보니 하나만 선택해도 active가 되는 것이 맞다고 합니다. 필터의 기능이 전체 중에 필터링 할 부분만 선택하는 기능이라고 생각합니다!

hamxxn reviewed (commented):
아직 유효성 검증이 안되고 있는 것 같아요!
확인 부탁드려요!

holdn2 commented:
> 아직 유효성 검증이 안되고 있는 것 같아요! 확인 부탁드려요!

어떤 유효성 검증인지 말씀해주세요!
